src/isomorphism/vf2_mapping.rs

```diff
@@ -0,0 +1,178 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+#![allow(clippy::too_many_arguments)]
+// This module was originally forked from petgraph's isomorphism module @ v0.5.0
+// to support StableGraph inputs instead of petgraph's generic Graph. However it has
+// since diverged significantly from the original petgraph implementation.
+
+use std::cmp::Ordering;
+use std::iter::Iterator;
+
+use pyo3::gc::PyVisit;
+use pyo3::prelude::*;
+use pyo3::PyTraverseError;
+
+use petgraph::stable_graph::{EdgeIndex, NodeIndex};
+use petgraph::EdgeType;
+use petgraph::{Directed, Undirected};
+
+use crate::iterators::NodeMap;
+use crate::StablePyGraph;
+
+use rustworkx_core::isomorphism::vf2;
+
+struct PyMatcher(Option<PyObject>);
+
+impl<Ty: EdgeType> vf2::NodeMatcher<StablePyGraph<Ty>, StablePyGraph<Ty>> for PyMatcher {
+    type Error = PyErr;
+
+    fn enabled(&self) -> bool {
+        self.0.is_some()
+    }
+
+    fn eq(
+        &mut self,
+        g0: &StablePyGraph<Ty>,
+        g1: &StablePyGraph<Ty>,
+        n0: NodeIndex,
+        n1: NodeIndex,
+    ) -> Result<bool, Self::Error> {
+        if let (Some(a), Some(b)) = (g0.node_weight(n0), g1.node_weight(n1)) {
+            unsafe {
```

Comment:
Just because PyO3 lets you do this does mean we should take it. I think it is a big amount of risk for a refactor